src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,175 @@
+     * @throws RuntimeException If an error occurs while reading the file.
+     */
+    private String[] readFromFile(String fileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            String value = reader.readLine();
+            while (value != null) {
+                stringBuilder.append(value).append(SPACE);
+                value = reader.readLine();
+            }
+            return prepareStatistic(stringBuilder.toString().split(SPACE));
+        } catch (IOException e) {
+            throw new RuntimeException(READ_DATA_EXCEPTION + fileName, e);
+        }
+    }
+
+    /**
+     * Writes the given content to the specified file.
+     *
+     * @param content  The content to write to the file.
+     * @param fileName The name of the file to write data to.
+     * @throws RuntimeException If an error occurs while writing to the file.
+     */
+    public void writeToFile(String[] content, String fileName) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(fileName))) {
+            for (String line : content) {
+                writer.write(line);
+                writer.newLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(WRITE_DATA_EXCEPTION + fileName, e);
+        }
+    }
+
+    /**
+     * Processes the raw data array and returns an array containing statistics.
+     *
+     * @param array The raw data array to process.
+     * @return An array containing processed statistics.
+     */
+    private String[] prepareStatistic(String[] array) {
+        int length = countUnicElements(array);
+        String[] operationList = new String[length];
+        int[] amountList = new int[length];
+
+        for (String row : array) {
+            String[] arrayElement = row.split(COMMA);
+            String operation = arrayElement[INDEX_ZERO];
+            int amount = Integer.parseInt(arrayElement[INDEX_ONE]);
+            int elementIndex = findMatch(operationList, operation);
+
+            if (elementIndex != -1) {
+                amountList[elementIndex] += amount;
+            } else {
+                if (SUPPLY.equals(operation)) {
+                    System.arraycopy(operationList, INDEX_ZERO, operationList,
+                            INDEX_ONE, length - INDEX_ONE);
+                    System.arraycopy(amountList, INDEX_ZERO, amountList,
+                            INDEX_ONE, length - INDEX_ONE);
+                    operationList[0] = operation;
+                    amountList[0] = amount;
+                } else {
+                    int emptyIndex = findMatch(operationList, null);
+                    operationList[emptyIndex] = operation;
+                    amountList[emptyIndex] = amount;
+                }
+            }
+        }
+        return concatArrays(operationList, amountList);
+    }
```

Comment:
This method is really big, but after a few hours of frustrating, it was the only solution I found to make sure that "supply" will be always in the first row...